docs/hugo/content/contributing/add-a-new-code-generated-resource.md

```diff
@@ -65,26 +62,32 @@ Once you have a working development environment, run the `task` command to run t
 Example:
 >  Replace cross-resource references with genruntime.ResourceReference: 
 > ["github.com/Azure/azure-service-operator/hack/generated/_apis/containerservice/v1alpha1api20210501/PrivateLinkResource.Id" looks like a resource reference but was not labelled as one. 
-> It might need to be manually added to `newKnownReferencesMap`,
 
-To fix this error, determine whether the property in question is an ARM ID or not, and then update the `newKnownReferencesMap` function 
-in [add_cross_resource_references.go](https://github.com/Azure/azure-service-operator/blob/main/v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go#:~:text=func-,newknownreferencesmap,-).
+To fix this error, determine whether the property in question is an ARM ID or not, and then update the `objectModelConfiguration` section in the configuration file.
 
-If the property is an ARM ID, update `newKnownReferencesMap` to flag that property as a reference:
-```go
-{
-       typeName: astmodel.MakeTypeName(configuration.MakeLocalPackageReference("containerservice", "v1alpha1api20210501"), "PrivateLinkResource"),
-       propName: "Id", 
-}: true,
+Find the section you added earlier, adding your property with an `$armReference:` declaration nested below.
+
+If the property is an ARM ID, use `$armReference: true` to flag that property as a reference:
+
+```yaml
+network:
+  2020-11-01:
+    NetworkSecurityGroup:
+      $export: true
+      PrivateLinkResource: 
+        $armReference: true // the property IS an ARM reference
```

Comment:
minor: In YAML the comments are `#`, aren't they?

Reply:
Fixed.